ionoscloud/data_source_dbaas_pgsql_cluster.go

```diff
@@ -0,0 +1,310 @@
+				Description: "The number of CPU cores per replica.",
+				Computed:    true,
+			},
+			"ram_size": {
+				Type:        schema.TypeString,
+				Description: "The amount of memory per replica.",
+				Computed:    true,
+			},
+			"storage_size": {
+				Type:        schema.TypeString,
+				Description: "The amount of storage per replica.",
+				Computed:    true,
+			},
+			"storage_type": {
+				Type:        schema.TypeString,
+				Description: "The storage type used in your cluster.",
+				Computed:    true,
+			},
+			"vdc_connections": {
+				Type:        schema.TypeList,
+				Description: "The VDC to connect to your cluster.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"vdc_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"lan_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"ip_address": {
+							Type:        schema.TypeString,
+							Description: "The IP and subnet for the database.\n          Note the following unavailable IP ranges:\n          10.233.64.0/18\n          10.233.0.0/18\n          10.233.114.0/24",
+							Computed:    true,
+						},
+					},
+				},
+			},
+			"location": {
+				Type:        schema.TypeString,
+				Description: "The physical location where the cluster will be created. This will be where all of your instances live. Property cannot be modified after datacenter creation (disallowed in update requests)",
+				Computed:    true,
+			},
+			"backup_enabled": {
+				Type:       schema.TypeBool,
+				Deprecated: "Deprecated: backup is always enabled.\n      Enables automatic backups of your cluster.",
+				Computed:   true,
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				Description: "a weekly 4 hour-long window, during which maintenance might occur",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"weekday": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+			"credentials": {
+				Type:        schema.TypeList,
+				Description: "Credentials for the database user to be created.",
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"username": {
+							Type:        schema.TypeString,
+							Description: "the username for the initial postgres user. some system usernames\n          are restricted (e.g. \"postgres\", \"admin\", \"standby\")",
+							Computed:    true,
+						},
+						"password": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func dataSourceDbaasPgSqlReadCluster(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).DbaasClient
```

Comment:
Lets externalize the read logic to be able to reuse it in the import section and in data sources.